### PR TITLE
Do not use idle event for token auth

### PR DESCRIPTION
**Pull request by ryanhamley**
## Launch Checklist

 - [x] briefly describe the changes in this PR
    - closes https://github.com/mapbox/gl-js-team/issues/73
 - [x] manually test the debug page
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

File: src/ui/map.js [hunk 1]
```diff
@@ -2640,6 +2636,10 @@ class Map extends Camera {
 
         if (this._loaded && !this._fullyLoaded && !somethingDirty) {
             this._fullyLoaded = true;
+            if (this._isInitialLoad) {
```
Comment (arindam1993) [inline src/ui/map.js:2639]: The _initialLoad boolean shouldn't be required because_fullyLoaded tracks the same state.

**Review by arindam1993 — CHANGES_REQUESTED** [review]

File: src/ui/map.js [hunk 2]
```diff
@@ -2640,6 +2636,10 @@ class Map extends Camera {
...
+                this._authenticate();
+            }
+            this._isInitialLoad = false;
```
Comment (asheemmamoowala) [inline src/ui/map.js:2642]: Moving this `_isInitialLoad` flip here means that fading will be active sooner than the first `idle` event. That may have an impact on load time.

Comment (ryanhamley) [reply src/ui/map.js]: I moved `_isInitialLoad` back to the `idle` event and just call `_authenticate` here

**Review by asheemmamoowala — APPROVED** [review]

**Comment by ryanhamley** [comment]
Closing in favor of https://github.com/mapbox/mapbox-gl-js/pull/10458 because something is messed up with the query tests on CI for this PR

**Pull request closed without merging**